### PR TITLE
47 tv discover

### PR DESCRIPTION
**Pull request by vnoelifant**
Mostly related to issue #22 and #59. Debugging the following error: 'ManyRelatedManager' object is not iterable when I clicked on "View Details" for Guardians of the Galaxy movie from the home page trending movies. Traceback is shown below. I am also trying to add the Videos model, but not sure how to associate this with the Movie model. An example videos data is also shown below. It looks like their is a common id that can link to a movie ID. 

```\Environment:


Request Method: GET
Request URL: http://127.0.0.1:8000/movie/447365/

Django Version: 4.1.2
Python Version: 3.9.17
Installed Applications:
['django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'bootstrap5',
 'movie_search.apps.MovieSearchConfig',
 'django_countries']
Installed Middleware:
['django.middleware.security.SecurityMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware']


Template error:
In template /home/vnoelifant/movie-search-app/templates/movie_detail.html, error at line 40
   'ManyRelatedManager' object is not iterable
   30 :     }
   31 : </style>
   32 : 
   33 : <div class="container first" id="detailBc"> 
   34 :     
   35 :     <img src="https://image.tmdb.org/t/p/w500{{ movie_detail.backdrop_path }}" alt="..." style="border-radius:15px; margin-left: 30px; margin-top: 30px;" >
   36 :     
   37 :     <div class="col mov-info" id="mediaInfo" >
   38 :         <h2> {{ movie_detail.title }}</h2>
   39 :         <h5><b>Genre:</b></h5>
   40 :          {% for genre in movie_detail.genres %} 
   41 :             <h8>{{genre.name}} |</h8> 
   42 :         {% endfor %}
   43 :         <h4><b>{{ movie_detail.tagline }}</b></h4>
   44 :         <h4>Total Votes: <b>{{movie_detail.vote_count}}</b> | Vote Score: <b>{{ movie_detail.vote_average }}</b> | Popularity: <b>{{movie_detail.popularity}}</b></h4>
   45 :         <h4>Release Date: <b>{{ movie_detail.release_date }}</b></h4> 
   46 :         <h4>Duration: <b>{{ movie_detail.runtime }} minutes</b></h4>
   47 :         <h4>Production Companies:</h4>
   48 :         {% for company in movie_detail.production_companies %}
   49 :             <h8>{{company.name}} |</h8>
   50 :         {% endfor %}


Traceback (most recent call last):
  File "/home/vnoelifant/.local/share/virtualenvs/movie-search-app-Z2wqbYPg/lib/python3.9/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
  File "/home/vnoelifant/.local/share/virtualenvs/movie-search-app-Z2wqbYPg/lib/python3.9/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/vnoelifant/movie-search-app/movie_search/views.py", line 318, in movie_detail
    return render(request, "movie_detail.html", context)
  File "/home/vnoelifant/.local/share/virtualenvs/movie-search-app-Z2wqbYPg/lib/python3.9/site-packages/django/shortcuts.py", line 24, in render
    content = loader.render_to_string(template_name, context, request, using=using)
  File "/home/vnoelifant/.local/share/virtualenvs/movie-search-app-Z2wqbYPg/lib/python3.9/site-packages/django/template/loader.py", line 62, in render_to_string
    return template.render(context, request)
  File "/home/vnoelifant/.local/share/virtualenvs/movie-search-app-Z2wqbYPg/lib/python3.9/site-packages/django/template/backends/django.py", line 62, in render
    return self.template.render(context)
  File "/home/vnoelifant/.local/share/virtualenvs/movie-search-app-Z2wqbYPg/lib/python3.9/site-packages/django/template/base.py", line 175, in render
    return self._render(context)
  File "/home/vnoelifant/.local/share/virtualenvs/movie-search-app-Z2wqbYPg/lib/python3.9/site-packages/django/template/base.py", line 167, in _render
    return self.nodelist.render(context)
  File "/home/vnoelifant/.local/share/virtualenvs/movie-search-app-Z2wqbYPg/lib/python3.9/site-packages/django/template/base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
  File "/home/vnoelifant/.local/share/virtualenvs/movie-search-app-Z2wqbYPg/lib/python3.9/site-packages/django/template/base.py", line 1005, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
  File "/home/vnoelifant/.local/share/virtualenvs/movie-search-app-Z2wqbYPg/lib/python3.9/site-packages/django/template/base.py", line 966, in render_annotated
    return self.render(context)
  File "/home/vnoelifant/.local/share/virtualenvs/movie-search-app-Z2wqbYPg/lib/python3.9/site-packages/django/template/loader_tags.py", line 157, in render
    return compiled_parent._render(context)
  File "/home/vnoelifant/.local/share/virtualenvs/movie-search-app-Z2wqbYPg/lib/python3.9/site-packages/django/template/base.py", line 167, in _render
    return self.nodelist.render(context)
  File "/home/vnoelifant/.local/share/virtualenvs/movie-search-app-Z2wqbYPg/lib/python3.9/site-packages/django/template/base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
  File "/home/vnoelifant/.local/share/virtualenvs/movie-search-app-Z2wqbYPg/lib/python3.9/site-packages/django/template/base.py", line 1005, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
  File "/home/vnoelifant/.local/share/virtualenvs/movie-search-app-Z2wqbYPg/lib/python3.9/site-packages/django/template/base.py", line 966, in render_annotated
    return self.render(context)
  File "/home/vnoelifant/.local/share/virtualenvs/movie-search-app-Z2wqbYPg/lib/python3.9/site-packages/django/template/loader_tags.py", line 63, in render
    result = block.nodelist.render(context)
  File "/home/vnoelifant/.local/share/virtualenvs/movie-search-app-Z2wqbYPg/lib/python3.9/site-packages/django/template/base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
  File "/home/vnoelifant/.local/share/virtualenvs/movie-search-app-Z2wqbYPg/lib/python3.9/site-packages/django/template/base.py", line 1005, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
  File "/home/vnoelifant/.local/share/virtualenvs/movie-search-app-Z2wqbYPg/lib/python3.9/site-packages/django/template/base.py", line 966, in render_annotated
    return self.render(context)
  File "/home/vnoelifant/.local/share/virtualenvs/movie-search-app-Z2wqbYPg/lib/python3.9/site-packages/django/template/defaulttags.py", line 193, in render
    values = list(values)

Exception Type: TypeError at /movie/447365/
Exception Value: 'ManyRelatedManager' object is not iterable
```

videos.json for movie "Lost in Translation" (ID 153)
```
{
    "id": 153,
    "results": [
        {
            "id": "5a8f380392514164660037c6",
            "iso_3166_1": "US",
            "iso_639_1": "en",
            "key": "nsf7cMjPa2I",
            "name": "Sofia Coppola interview on \"Lost in Translation\" (2003)",
            "official": false,
            "published_at": "2017-02-20T22:49:12.000Z",
            "site": "YouTube",
            "size": 720,
            "type": "Featurette"
        },
        {
            "id": "5a8f3ab60e0a2639ae003c8f",
            "iso_3166_1": "US",
            "iso_639_1": "en",
            "key": "LSsNbFwePAE",
            "name": "Lost In Translation Intro Scene",
            "official": false,
            "published_at": "2015-09-22T11:26:18.000Z",
            "site": "YouTube",
            "size": 1080,
            "type": "Clip"
        },
        {
            "id": "5a8f3c430e0a2639ae003df8",
            "iso_3166_1": "US",
            "iso_639_1": "en",
            "key": "lpOdAHwRnXY",
            "name": "Lost in Translation (10/10) Movie CLIP - A Secret Goodbye (2003) HD",
            "official": false,
            "published_at": "2011-05-30T11:44:47.000Z",
            "site": "YouTube",
            "size": 720,
            "type": "Clip"
        },
        {
            "id": "5a8f3c22c3a3682db3003a9d",
            "iso_3166_1": "US",
            "iso_639_1": "en",
            "key": "i86FbeyKK1U",
            "name": "Lost in Translation (5/10) Movie CLIP - Ditzy Actress (2003) HD",
            "official": false,
            "published_at": "2011-05-30T11:00:31.000Z",
            "site": "YouTube",
            "size": 720,
            "type": "Clip"
        },
        {
            "id": "5a8f3c02c3a3682de60037ab",
            "iso_3166_1": "US",
            "iso_639_1": "en",
            "key": "wixLlPFJgyQ",
            "name": "Lost in Translation (4/10) Movie CLIP - Bad Exercise (2003) HD",
            "official": false,
            "published_at": "2011-05-30T10:47:32.000Z",
            "site": "YouTube",
            "size": 720,
            "type": "Clip"
        },
        {
            "id": "5af20ae8c3a36855b600321c",
            "iso_3166_1": "US",
            "iso_639_1": "en",
            "key": "1s4YQqnbnM8",
            "name": "Lost in Translation - Trailer",
            "official": true,
            "published_at": "2011-04-11T21:48:31.000Z",
            "site": "YouTube",
            "size": 480,
            "type": "Trailer"
        },
        {
            "id": "5a8f3c66c3a3682de6003801",
            "iso_3166_1": "US",
            "iso_639_1": "en",
            "key": "pLXJhTDVoZQ",
            "name": "Lost in translation, Karaoke",
            "official": false,
            "published_at": "2010-11-03T05:00:22.000Z",
            "site": "YouTube",
            "size": 480,
            "type": "Clip"
        }
    ]
}
```

### CODE REVIEW

**Comment by vnoelifant** [comment]
> Mostly related to issue #22 and #59. Debugging the following error: 'ManyRelatedManager' object is not iterable when I clicked on "View Details" for Guardians of the Galaxy movie from the home page trending movies. Traceback is shown below. I am also trying to add the Videos model, but not sure how to associate this with the Movie model. An example videos data is also shown below. It looks like their is a common id that can link to a movie ID.
> 
> ```
> 
> 
> Request Method: GET
> Request URL: http://127.0.0.1:8000/movie/447365/
> 
> Django Version: 4.1.2
> Python Version: 3.9.17
> Installed Applications:
> ['django.contrib.admin',
>  'django.contrib.auth',
>  'django.contrib.contenttypes',
>  'django.contrib.sessions',
>  'django.contrib.messages',
>  'django.contrib.staticfiles',
>  'bootstrap5',
>  'movie_search.apps.MovieSearchConfig',
>  'django_countries']
> Installed Middleware:
> ['django.middleware.security.SecurityMiddleware',
>  'django.contrib.sessions.middleware.SessionMiddleware',
>  'django.middleware.common.CommonMiddleware',
>  'django.middleware.csrf.CsrfViewMiddleware',
>  'django.contrib.auth.middleware.AuthenticationMiddleware',
>  'django.contrib.messages.middleware.MessageMiddleware',
>  'django.middleware.clickjacking.XFrameOptionsMiddleware']
> 
> 
> Template error:
> In template /home/vnoelifant/movie-search-app/templates/movie_detail.html, error at line 40
>    'ManyRelatedManager' object is not iterable
>    30 :     }
>    31 : </style>
>    32 : 
>    33 : <div class="container first" id="detailBc"> 
>    34 :     
>    35 :     
>    36 :     
>    37 :     <div class="col mov-info" id="mediaInfo" >
>    38 :         <h2> {{ movie_detail.title }}</h2>
>    39 :         <h5><b>Genre:</b></h5>
>    40 :          {% for genre in movie_detail.genres %} 
>    41 :             <h8>{{genre.name}} |</h8> 
>    42 :         {% endfor %}
>    43 :         <h4><b>{{ movie_detail.tagline }}</b></h4>
>    44 :         <h4>Total Votes: <b>{{movie_detail.vote_count}}</b> | Vote Score: <b>{{ movie_detail.vote_average }}</b> | Popularity: <b>{{movie_detail.popularity}}</b></h4>
>    45 :         <h4>Release Date: <b>{{ movie_detail.release_date }}</b></h4> 
>    46 :         <h4>Duration: <b>{{ movie_detail.runtime }} minutes</b></h4>
>    47 :         <h4>Production Companies:</h4>
>    48 :         {% for company in movie_detail.production_companies %}
>    49 :             <h8>{{company.name}} |</h8>
>    50 :         {% endfor %}
> 
> 
> Traceback (most recent call last):
>   File "/home/vnoelifant/.local/share/virtualenvs/movie-search-app-Z2wqbYPg/lib/python3.9/site-packages/django/core/handlers/exception.py", line 55, in inner
>     response = get_response(request)
>   File "/home/vnoelifant/.local/share/virtualenvs/movie-search-app-Z2wqbYPg/lib/python3.9/site-packages/django/core/handlers/base.py", line 197, in _get_response
>     response = wrapped_callback(request, *callback_args, **callback_kwargs)
>   File "/home/vnoelifant/movie-search-app/movie_search/views.py", line 318, in movie_detail
>     return render(request, "movie_detail.html", context)
>   File "/home/vnoelifant/.local/share/virtualenvs/movie-search-app-Z2wqbYPg/lib/python3.9/site-packages/django/shortcuts.py", line 24, in render
>     content = loader.render_to_string(template_name, context, request, using=using)
>   File "/home/vnoelifant/.local/share/virtualenvs/movie-search-app-Z2wqbYPg/lib/python3.9/site-packages/django/template/loader.py", line 62, in render_to_string
>     return template.render(context, request)
>   File "/home/vnoelifant/.local/share/virtualenvs/movie-search-app-Z2wqbYPg/lib/python3.9/site-packages/django/template/backends/django.py", line 62, in render
>     return self.template.render(context)
>   File "/home/vnoelifant/.local/share/virtualenvs/movie-search-app-Z2wqbYPg/lib/python3.9/site-packages/django/template/base.py", line 175, in render
>     return self._render(context)
>   File "/home/vnoelifant/.local/share/virtualenvs/movie-search-app-Z2wqbYPg/lib/python3.9/site-packages/django/template/base.py", line 167, in _render
>     return self.nodelist.render(context)
>   File "/home/vnoelifant/.local/share/virtualenvs/movie-search-app-Z2wqbYPg/lib/python3.9/site-packages/django/template/base.py", line 1005, in render
>     return SafeString("".join([node.render_annotated(context) for node in self]))
>   File "/home/vnoelifant/.local/share/virtualenvs/movie-search-app-Z2wqbYPg/lib/python3.9/site-packages/django/template/base.py", line 1005, in <listcomp>
>     return SafeString("".join([node.render_annotated(context) for node in self]))
>   File "/home/vnoelifant/.local/share/virtualenvs/movie-search-app-Z2wqbYPg/lib/python3.9/site-packages/django/template/base.py", line 966, in render_annotated
>     return self.render(context)
>   File "/home/vnoelifant/.local/share/virtualenvs/movie-search-app-Z2wqbYPg/lib/python3.9/site-packages/django/template/loader_tags.py", line 157, in render
>     return compiled_parent._render(context)
>   File "/home/vnoelifant/.local/share/virtualenvs/movie-search-app-Z2wqbYPg/lib/python3.9/site-packages/django/template/base.py", line 167, in _render
>     return self.nodelist.render(context)
>   File "/home/vnoelifant/.local/share/virtualenvs/movie-search-app-Z2wqbYPg/lib/python3.9/site-packages/django/template/base.py", line 1005, in render
>     return SafeString("".join([node.render_annotated(context) for node in self]))
>   File "/home/vnoelifant/.local/share/virtualenvs/movie-search-app-Z2wqbYPg/lib/python3.9/site-packages/django/template/base.py", line 1005, in <listcomp>
>     return SafeString("".join([node.render_annotated(context) for node in self]))
>   File "/home/vnoelifant/.local/share/virtualenvs/movie-search-app-Z2wqbYPg/lib/python3.9/site-packages/django/template/base.py", line 966, in render_annotated
>     return self.render(context)
>   File "/home/vnoelifant/.local/share/virtualenvs/movie-search-app-Z2wqbYPg/lib/python3.9/site-packages/django/template/loader_tags.py", line 63, in render
>     result = block.nodelist.render(context)
>   File "/home/vnoelifant/.local/share/virtualenvs/movie-search-app-Z2wqbYPg/lib/python3.9/site-packages/django/template/base.py", line 1005, in render
>     return SafeString("".join([node.render_annotated(context) for node in self]))
>   File "/home/vnoelifant/.local/share/virtualenvs/movie-search-app-Z2wqbYPg/lib/python3.9/site-packages/django/template/base.py", line 1005, in <listcomp>
>     return SafeString("".join([node.render_annotated(context) for node in self]))
>   File "/home/vnoelifant/.local/share/virtualenvs/movie-search-app-Z2wqbYPg/lib/python3.9/site-packages/django/template/base.py", line 966, in render_annotated
>     return self.render(context)
>   File "/home/vnoelifant/.local/share/virtualenvs/movie-search-app-Z2wqbYPg/lib/python3.9/site-packages/django/template/defaulttags.py", line 193, in render
>     values = list(values)
> 
> Exception Type: TypeError at /movie/447365/
> Exception Value: 'ManyRelatedManager' object is not iterable
> ```
> 
> videos.json for movie "Lost in Translation" (ID 153)
> 
> ```
> {
>     "id": 153,
>     "results": [
>         {
>             "id": "5a8f380392514164660037c6",
>             "iso_3166_1": "US",
>             "iso_639_1": "en",
>             "key": "nsf7cMjPa2I",
>             "name": "Sofia Coppola interview on \"Lost in Translation\" (2003)",
>             "official": false,
>             "published_at": "2017-02-20T22:49:12.000Z",
>             "site": "YouTube",
>             "size": 720,
>             "type": "Featurette"
>         },
>         {
>             "id": "5a8f3ab60e0a2639ae003c8f",
>             "iso_3166_1": "US",
>             "iso_639_1": "en",
>             "key": "LSsNbFwePAE",
>             "name": "Lost In Translation Intro Scene",
>             "official": false,
>             "published_at": "2015-09-22T11:26:18.000Z",
>             "site": "YouTube",
>             "size": 1080,
>             "type": "Clip"
>         },
>         {
>             "id": "5a8f3c430e0a2639ae003df8",
>             "iso_3166_1": "US",
>             "iso_639_1": "en",
>             "key": "lpOdAHwRnXY",
>             "name": "Lost in Translation (10/10) Movie CLIP - A Secret Goodbye (2003) HD",
>             "official": false,
>             "published_at": "2011-05-30T11:44:47.000Z",
>             "site": "YouTube",
>             "size": 720,
>             "type": "Clip"
>         },
>         {
>             "id": "5a8f3c22c3a3682db3003a9d",
>             "iso_3166_1": "US",
>             "iso_639_1": "en",
>             "key": "i86FbeyKK1U",
>             "name": "Lost in Translation (5/10) Movie CLIP - Ditzy Actress (2003) HD",
>             "official": false,
>             "published_at": "2011-05-30T11:00:31.000Z",
>             "site": "YouTube",
>             "size": 720,
>             "type": "Clip"
>         },
>         {
>             "id": "5a8f3c02c3a3682de60037ab",
>             "iso_3166_1": "US",
>             "iso_639_1": "en",
>             "key": "wixLlPFJgyQ",
>             "name": "Lost in Translation (4/10) Movie CLIP - Bad Exercise (2003) HD",
>             "official": false,
>             "published_at": "2011-05-30T10:47:32.000Z",
>             "site": "YouTube",
>             "size": 720,
>             "type": "Clip"
>         },
>         {
>             "id": "5af20ae8c3a36855b600321c",
>             "iso_3166_1": "US",
>             "iso_639_1": "en",
>             "key": "1s4YQqnbnM8",
>             "name": "Lost in Translation - Trailer",
>             "official": true,
>             "published_at": "2011-04-11T21:48:31.000Z",
>             "site": "YouTube",
>             "size": 480,
>             "type": "Trailer"
>         },
>         {
>             "id": "5a8f3c66c3a3682de6003801",
>             "iso_3166_1": "US",
>             "iso_639_1": "en",
>             "key": "pLXJhTDVoZQ",
>             "name": "Lost in translation, Karaoke",
>             "official": false,
>             "published_at": "2010-11-03T05:00:22.000Z",
>             "site": "YouTube",
>             "size": 480,
>             "type": "Clip"
>         }
>     ]
> }
> ```

@bbelderbos Okay I think I fixed it with adding ".all" in template tag. https://stackoverflow.com/questions/70927390/how-do-i-iterate-manytomany-field-in-django-template-tag

**Comment by bbelderbos** [comment]
First error I get going to that movie is actually this?

image

**Comment by bbelderbos** [comment]
That's easy to fix, I will get past this and review a bit more tomorrow, today was busy.

**Comment by bbelderbos** [comment]
hm second visit, it's probably cached, then it works:

image

but then hitting another movie:

image

that does not happen on your end?

I did have to do a merge conflict so let me maybe set up the project again to make sure I have your exact code ...

**Comment by bbelderbos** [comment]
Nope, doing a fresh install and checkout of branch I still get this error going to Guardians of the Galaxy Vol. 3

image

Upon first click, refresh the page and it works.

File: movie_search/media_api.py [hunk 1]
```diff
@@ -87,17 +85,15 @@ def get_media_data(
...
-    print("Params: ", params)
-
     response = requests.get(url, params=params)
 
     return response.json()
 
 def main():
     from utils import dump_movie_data_to_json
-    data = get_media_data("/watch/providers/tv")
+    data = get_media_data("/movie/153/recommendations")
```
Comment (bbelderbos) [inline movie_search/media_api.py:94]: should this 153 be hardcoded?

File: movie_search/models.py [hunk 2]
```diff
@@ -18,3 +21,35 @@ class Provider(models.Model):
...
         return f"{self.provider_id}: {self.name}"
+
+
+class Video(models.Model):
+   
+    name = models.CharField(max_length=200)
+    key =  models.CharField(max_length=200)
+
+    def __str__(self):
+        """String for representing the Model object."""
```
Comment (bbelderbos) [inline movie_search/models.py:32]: I typically don't do a docstring on this method, should be clear

File: movie_search/models.py [hunk 3]
```diff
@@ -18,3 +21,35 @@ class Provider(models.Model):
...
+        return f"{self.name}: {self.key}"
+
+class Movie(models.Model):
+    movie_id = models.PositiveSmallIntegerField(default=0)
+    title = models.CharField(max_length=200, null=True, blank=True)
```
Comment (bbelderbos) [inline movie_search/models.py:37]: are all these fields optional?

File: movie_search/models.py [hunk 4]
```diff
@@ -18,3 +21,35 @@ class Provider(models.Model):
...
+    vote_average = models.PositiveSmallIntegerField(default=0)
+    popularity = models.IntegerField(default=0, null=True, blank=True)
+    release_date = models.CharField(max_length=200, null=True, blank=True)
+    runtime = models.IntegerField(default=0, null=True, blank=True)
+    production_company = models.CharField(max_length=200, null=True, blank=True)
+    overview = models.TextField(default="", null=True, blank=True)
+    budget = models.IntegerField(default=0, null=True, blank=True)
+    revenue = models.IntegerField(default=0, null=True, blank=True)
+    homepage = models.CharField(max_length=200, null=True, blank=True)
+    videos = models.ForeignKey(Video, on_delete=models.CASCADE,null=True, blank=True)
```
Comment (bbelderbos) [inline movie_search/models.py:51]: I would call this `video`

File: movie_search/models.py [hunk 5]
```diff
@@ -18,3 +21,35 @@ class Provider(models.Model):
...
+
+    def __str__(self):
+        """String for representing the Model object."""
```
Comment (bbelderbos) [inline movie_search/models.py:54]: remove

File: movie_search/models.py [hunk 6]
```diff
@@ -18,3 +21,35 @@ class Provider(models.Model):
...
+    tagline = models.CharField(max_length=200, null=True, blank=True)
+    vote_count = models.PositiveSmallIntegerField(default=0)
+    vote_average = models.PositiveSmallIntegerField(default=0)
+    popularity = models.IntegerField(default=0, null=True, blank=True)
+    release_date = models.CharField(max_length=200, null=True, blank=True)
+    runtime = models.IntegerField(default=0, null=True, blank=True)
+    production_company = models.CharField(max_length=200, null=True, blank=True)
+    overview = models.TextField(default="", null=True, blank=True)
+    budget = models.IntegerField(default=0, null=True, blank=True)
+    revenue = models.IntegerField(default=0, null=True, blank=True)
```
Comment (bbelderbos) [inline movie_search/models.py:49]: these are whole numbers, right? no float?

File: movie_search/models.py [hunk 7]
```diff
@@ -18,3 +21,35 @@ class Provider(models.Model):
...
+
+class Movie(models.Model):
+    movie_id = models.PositiveSmallIntegerField(default=0)
+    title = models.CharField(max_length=200, null=True, blank=True)
+    backdrop_path = models.CharField(max_length=200, null=True, blank=True)
+    genres = models.ManyToManyField(Genre, related_name="movies", blank=True)
+    tagline = models.CharField(max_length=200, null=True, blank=True)
+    vote_count = models.PositiveSmallIntegerField(default=0)
+    vote_average = models.PositiveSmallIntegerField(default=0)
+    popularity = models.IntegerField(default=0, null=True, blank=True)
```
Comment (bbelderbos) [inline movie_search/models.py:43]: is this 1-10? then you could add something like `validators=[MaxValueValidator(10), MinValueValidator(1)]`

File: movie_search/views.py [hunk 8]
```diff
@@ -240,20 +241,82 @@ def search(request):
...
-def movie_detail(request, obj_id):
-
-    movie_detail = media_api.get_media_data(f"/movie/{obj_id}")
-    # pprint("MOVIE DETAIL: ", movie_detail)
+    genres = movie.get("genres")
 
-    movie_videos = media_api.get_media_data(f"/movie/{obj_id}/videos")
+    if genres is not None:
+        for row in genres:
+            genre, inserted = Genre.objects.get_or_create(
```
Comment (bbelderbos) [inline movie_search/views.py:252]: this create objects in a loop, you could do one bulk create as well

File: movie_search/views.py [hunk 9]
```diff
@@ -240,20 +241,82 @@ def search(request):
...
-    context = {
-        "movie_detail": movie_detail,
-        "movie_videos": movie_videos,
-        "type": "movie",
-    }
+def movie_detail(request, obj_id):
 
+    try:
+        movie_detail = Movie.objects.get(movie_id=obj_id)
+        context = {
```
Comment (bbelderbos) [inline movie_search/views.py:264]: the context dict below has more items, do you need those here as well (in case of the success scenario)?

**Review by bbelderbos — APPROVED** [review]
Good work, some feedback. I don't see any code related to the issue?

File: movie_search/views.py (hunk 8 above)
Comment (vnoelifant) [reply movie_search/views.py]: So I tried adding this:
```
def get_movie_genres(movie):
    
    movie_genres = []

    genres = movie.get("genres")

    if genres is not None:
        for row in genres:
            genre = Genre(name=row.get("name", ""),genre_id=row.get("id", ""),)
            movie_genres.append(genre)
        Genre.objects.bulk_create(movie_genres)
    
    return movie_genres
```

But I encountered this error:

image

I also have fixtures located in https://github.com/vnoelifant/movie-search-app/tree/47-tv-discover/movie_search/fixtures for the genres and the movie providers. I am confused if I need to load this if doing a Django create/get_or_create.

**Comment by vnoelifant** [comment]
> Nope, doing a fresh install and checkout of branch I still get this error going to Guardians of the Galaxy Vol. 3
> 
> image
> Upon first click, refresh the page and it works.

I have been getting this too, but not for every movie. I am not sure why.

File: movie_search/media_api.py (hunk 1 above)
Comment (vnoelifant) [reply movie_search/media_api.py:94]: That was only for testing when running media_api.py directly. I run media_api.py directly when I want to inspect the JSON data more.

File: movie_search/views.py (hunk 8 above)
Comment (bbelderbos) [reply movie_search/views.py]: Seems issue with database setting, the code looks good. Let's reproduce and debug this today ...

**Comment by bbelderbos** [comment]
> > Nope, doing a fresh install and checkout of branch I still get this error going to Guardians of the Galaxy Vol. 3
> > image
> > Upon first click, refresh the page and it works.
> 
> I have been getting this too, but not for every movie. I am not sure why.

Yes this was working before. Let's debug together today ...

File: movie_search/models.py (hunk 6 above)
Comment (vnoelifant) [reply movie_search/models.py:46]: > these are whole numbers, right? no float?

They are integers per the movie detail response https://developer.themoviedb.org/reference/movie-details. Is this correct syntax?

File: movie_search/models.py (hunk 7 above)
Comment (vnoelifant) [reply movie_search/models.py:40]: vote count is an integer, and vote average and popularity are termed "numbers" in the API docs. Here is an example response. I think vote average and popularity should change to models.FloatField() right?
```
{  
  "poster_path": "/IfB9hy4JH1eH6HEfIgIGORXi5h.jpg",  
  "adult": false,  
  "overview": "Jack Reacher must uncover the truth behind a major government conspiracy in order to clear his name. On the run as a fugitive from the law, Reacher uncovers a potential secret from his past that could change his life forever.",  
  "release_date": "2016-10-19",  
  "genre_ids": [  
    53,  
    28,  
    80,  
    18,  
    9648  
  ],  
  "id": 343611,  
  "original_title": "Jack Reacher: Never Go Back",  
  "original_language": "en",  
  "title": "Jack Reacher: Never Go Back",  
  "backdrop_path": "/4ynQYtSEuU5hyipcGkfD6ncwtwz.jpg",  
  "popularity": 26.818468,  
  "vote_count": 201,  
  "video": false,  
  "vote_average": 4.19  
}
```

Comment (bbelderbos) [reply movie_search/models.py:40]: oh then my suggestion does not match, yes seems more like `models.FloatField()`s then.

File: movie_search/models.py (hunk 6 above)
Comment (bbelderbos) [reply movie_search/models.py:46]: I see it now there, yes then we're good on these column types.

**Pull request merged**